devicetypes/widomsrl/widom-smart-dry-contact.src/widom-smart-dry-contact.groovy

```diff
@@ -0,0 +1,353 @@
+/**
+ *  Widom Smart DRY contact
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *	  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+metadata {
+	definition (name: "WiDom Smart Dry Contact", namespace: "WiDomsrl", author: "WiDom srl", ocfDeviceType: "oic.d.switch", mnmn: "SmartThings", vid: "generic-switch") {
+		capability "Switch"
+		capability "Configuration"
+		capability "Health Check"
+
+		fingerprint mfr: "0149", prod: "1214", model: "0900", deviceJoinName: "WiDom Dry Contact" // WiDom Dry Contact allow to turn On/Off a load with indipendent power supply.
+	}
+    
+	preferences {
+		input (
+			title: "WiDom Smart Dry Contact manual",
+			description: "Tap to view the manual.",
+			image: "https://www.widom.it/wp-content/uploads/2019/03/widom-3d-smart-dry-contact.gif",
+			url: "https://www.widom.it/wp-content/uploads/2020/04/Widom_Dry_Contact_IT_070420.pdf",
+			type: "href",
+			element: "href"
+		)
+
+		parameterMap().each {
+			input (
+					title: "${it.num}. ${it.title}",
+					description: it.descr,
+					type: "paragraph",
+					element: "paragraph"
+			)
+
+			input (
+					name: it.key,
+					title: null,
+					//description: "Default: $it.def" ,
+					type: it.type,
+					options: it.options,
+					//range: (it.min != null && it.max != null) ? "${it.min}..${it.max}" : null,
+					defaultValue: it.def,
+					required: false
+			)
+		}
+		input ( name: "logging", title: "Logging", type: "boolean", required: false )
+	}
+}
+
```

Comment:
Please remove unnecessary double line breaks, one line break is sufficient (applies to a whole file).

Reply:
fixed